roofit/roofitcore/test/testRooSimultaneous.cxx

```diff
@@ -456,3 +456,52 @@ TEST(RooSimultaneous, DuplicateExtendedPdfs)
    EXPECT_FLOAT_EQ(simPdfVal, 0.05);
    EXPECT_DOUBLE_EQ(simPdfVal, simPdfRef.getVal(normSet));
 }
+
+/// GitHub issue #19166.
+/// RooSimultaneous::fitTo() should work well with ConditionalOvservables.
+/// We test this by performing the similar test to a test case in
+/// RooSimultaneousSingleChannelCrossCheck.
+TEST_P(TestStatisticTest, RooSimultaneousSingleChannelCrossCheckWithCondVar)
+{
+   using namespace RooFit;
+
+   // silence log output
+   RooHelpers::LocalChangeMsgLevel changeMsgLvl(RooFit::WARNING);
+
+   RooWorkspace ws;
+   ws.factory("Uniform::uniform(width[1, 3])");
+   ws.factory("expr::sigma(\"@0*@1\", width, width_scale[0.8, 0.5, 1.5])");
+   ws.factory("Gaussian::model(x[-20, 20], mean[3, -20., 20.], sigma)");
+
+   RooRealVar &width = *ws.var("width");
+   RooAbsPdf &widthModel = *ws.pdf("uniform");
+   std::unique_ptr<RooDataSet> protoData{widthModel.generate(width, 100)};
+
+   RooRealVar &x = *ws.var("x");
+   RooAbsPdf &model = *ws.pdf("model");
+
+   std::unique_ptr<RooDataSet> data{model.generate(x, *protoData)};
+
+   RooCategory cat("cat", "cat");
+   cat.defineType("physics");
+
+   RooArgSet params;
+   RooArgSet initialParams;
+   model.getParameters(data->get(), params);
+   params.snapshot(initialParams);
+
+   RooSimultaneous modelSim("modelSim", "modelSim", {{"physics", &model}}, cat);
+
+   RooDataSet combData("combData", "combData", {x, width}, Index(cat), Import({{"physics", data.get()}}));
+
+   params.assign(initialParams);
+   std::unique_ptr<RooFitResult> resDirect{
+      model.fitTo(*data, RooFit::ConditionalObservables(width), RooFit::Save(), RooFit::PrintLevel(-1))};
+
+   params.assign(initialParams);
+   std::unique_ptr<RooFitResult> resSimWrapped{
+      modelSim.fitTo(combData, RooFit::ConditionalObservables(width), RooFit::Save(), RooFit::PrintLevel(-1))};
```

Comment:
```suggestion
   using namespace RooFit;

   params.assign(initialParams);
   std::unique_ptr<RooFitResult> resDirect{
      model.fitTo(*data, ConditionalObservables(width), Save(), PrintLevel(-1), _evalBackend)};

   params.assign(initialParams);
   std::unique_ptr<RooFitResult> resSimWrapped{
      modelSim.fitTo(combData, ConditionalObservables(width), Save(), PrintLevel(-1), _evalBackend)};
```

This is a parametrized test with different choices of the NLL evaluation backend, so you need to use this `_evalBackend` argument like in the other tests. Like this, the test will also run with the CUDA and legacy backends.